### PR TITLE
feat: Automatically hide window when losing focus

### PR DESCRIPTION
**Pull request by droid-Q**
## Description
Automatically hide window when losing focus

### Checklist:

* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you linted your code locally prior to submission?
* [x] Have you successfully ran app with your changes locally?

### CODE REVIEW

**Comment by agalwood** [comment]
I'm a little curious why there is such a need for such an auto-hide window. What kind of scene is needed?

**Comment by agalwood** [comment]
I think the configuration of `hide-window` should be changed to `auto-hide-window`, so the text and translation have to be redone.
The position of the auto hide window setting in the preferences should be moved to the appearance group.

**Comment by droid-Q** [comment]
> I'm a little curious why there is such a need for such an auto-hide window. What kind of scene is needed?

I just don't like to close the window by clicking the button.

**Comment by droid-Q** [comment]
> I think the configuration of `hide-window` should be changed to `auto-hide-window`, so the text and translation have to be redone.
> The position of the auto hide window setting in the preferences should be moved to the appearance group.

Ok, i will change it.

File: src/renderer/components/Preference/Basic.vue [hunk 1]
```diff
@@ -67,11 +67,11 @@
           </el-col>
         </el-form-item>
         <el-form-item
```
Comment (agalwood) [inline src/renderer/components/Preference/Basic.vue]: L69-77
The position of the auto hide window setting in the preferences should be moved to the appearance group.

**Comment by agalwood** [comment]
Thanks to your PR, I found a potential memory leak code : P
I have fixed it： https://github.com/agalwood/Motrix/commit/29b0d90bc716b2a6f58ad34bac2aa4c7b964e98f

**Comment by droid-Q** [comment]
> Thanks to your PR, I found a potential memory leak code : P
> I have fixed it： [29b0d90](https://github.com/agalwood/Motrix/commit/29b0d90bc716b2a6f58ad34bac2aa4c7b964e98f)

ok, thanks.

**Pull request merged**